### PR TITLE
Configure dataset size in demo script and notebook (fixes #273)

### LINKED ISSUE #273: Expose maximum model size parameter in CLI demo
Currently we only retrieve models of size 3GB or less in the CLI demo. It would be good to make this parameter configurable, as some people might want to test larger models. We could also possibly give sizes of popular models as a reference (e.g. gpt2, t5, llama).

### PR DESCRIPTION
**Pull request by viswavi**
# Description

This PR makes it easy to set the maximum model size (in both the demo script and the demo notebook). This PR also fixes a one-line bug in the dataset retriever, for convenience.

Example of how to set the maximum model size in the demo script:
Screenshot 2023-09-21 at 2 30 51 PM

# References

N/A

# Blocked by

N/A

### CODE REVIEW

**Review by neubig — COMMENTED** [review]
This looks great! It would be nice to mention the model sizes in the markdown comments for the jupyter notebook as well though.

File: prompt2model_demo.ipynb [hunk 1]
```diff
@@ -183,6 +183,7 @@
     "    model_descriptions_index_path=\"huggingface_data/huggingface_models/model_info/\",\n",
     "    use_bm25=True,\n",
     "    use_HyDE=True,\n",
+    "    model_size_limit_bytes=3e9,\n",
```
[the remark this answers is not in the document]
Comment (viswavi) [reply prompt2model_demo.ipynb:189]: Yes, done.

**Comment by viswavi** [comment]
@neubig I updated the markdown at this block to be clearer about what this parameter is.

File: prompt2model_demo.py [hunk 2]
```diff
@@ -78,6 +78,45 @@ def print_logo():
...
+        "B": ["b", "bytes"],
+        "KB": ["Kb", "kb", "kilobytes"],
+        "MB": ["Mb", "mb", "megabytes"],
+        "GB": ["Gb", "gb", "gigabytes"],
+        "TB": ["Tb", "tb", "terabytes"],
+        "PB": ["Pb", "pb", "petabytes"],
+    }
+    unit_matched = False
+    for unit, disambiguations in unit_disambiguations.items():
+        all_disambiguations = [unit] + disambiguations
```
Comment (hitenvidhani) [inline prompt2model_demo.py:101]: Cleaner implementation:
for unit, disambiguations in unit_disambiguations.items():
            if unit == key or unit in disambiguations:
                unit = key
                break

Comment (viswavi) [reply prompt2model_demo.py]: Done, thanks for the suggestion!

**Review by neubig — APPROVED** [review]

**Pull request merged**